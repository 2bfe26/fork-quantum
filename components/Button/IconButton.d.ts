import { ComponentType } from 'react';

export interface IconButtonProps {
    size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
    skin?: 'neutral' | 'primary' | 'secondary' | 'success' | 'warning' | 'error';
    theme?: {
        baseFontSize?: number;
        colors?: {};
        spacing?: {};
        components?: {
            button?: {};
        };
    };
}

type IconButton = ComponentType<IconButtonProps>;

export default IconButton;
