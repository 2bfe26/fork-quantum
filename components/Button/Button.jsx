import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import SocialButtons from './SocialButtons';
import { SIZES } from '../Grid/sub-components/shared/grid-config';
import theme from '../shared/theme';
import skins from './skins';
import Icon from '../Icon/Icon';

const fontSize = ({ size }) => {
  const sizes = {
    xsmall: '12px',
    small: '12px',
    medium: '16px',
    big: '20px',
    xbig: '24px',
  };
  return `font-size: ${sizes[size] || sizes.medium};`;
};

const iconSize = ({ size }) => {
  const sizes = {
    normal: '24px',
  };

  return `font-size: ${sizes[size] || sizes.normal};`;
};

const iconMargin = ({ size }) => {
  const margins = {
    normal: '-5px 5px -7px -5px',
  };

  return `margin: ${margins[size] || margins.normal};`;
};

const padding = ({ size }) => {
  const paddings = {
    xsmall: '4px 12px',
    small: '8px 12px',
    medium: '10px 12px',
    big: '11px 16px',
    xbig: '13px 16px',
  };

  return `padding: ${paddings[size] || paddings.normal};`;
};

const ButtonIcon = styled(Icon)`
  ${iconMargin};
`;

const StyledButton = styled.button`
  border-radius: ${theme.sizes.radius};
  font-weight: bold;
  text-align: center;
  text-decoration: ${props => (props.link ? 'underline' : 'none')};

  ${fontSize}
  ${padding}
  ${theme.mixins.transition()};

  ${ButtonIcon} {
    ${iconSize}
  }

  ${props =>
    props.full &&
    `
    width: 100%;
  `}

  ${props =>
    props.center &&
    `
    display: block;
    margin-left: auto;
    margin-right: auto;
  `}

  ${props => `
    cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  `}

  ${props => {
    const { unselected, selected, disabled, shadow } = skins(props);

    return `
      background-color: ${
        props.disabled ? disabled.background : unselected.background
      };
      border: 1px solid ${props.disabled ? disabled.border : unselected.border};
      color: ${props.disabled ? disabled.color : unselected.color};
      font-weight: ${
        props.disabled ? disabled.fontWeight : unselected.fontWeight
      };

      ${shadow ? theme.mixins.shadow() : undefined};

      &:active {
        ${shadow && theme.mixins.shadow(2)};
        background-color: ${selected.background};
        border-color: ${selected.border};
        color: ${selected.color};
      }
    `;
  }}

  @media (max-width: ${SIZES.phone}px) {
    margin-left: 0;
    margin-right: 0;
    width: 100%;
  }
`;

class Button extends React.Component {
  static Facebook = SocialButtons.Facebook;

  static Google = SocialButtons.Google;

  render() {
    const { children, icon, size, ...rest } = this.props;

    return (
      <StyledButton {...rest} size={size}>
        {icon && <ButtonIcon size={size} name={icon} />}
        {children}
      </StyledButton>
    );
  }
}

Button.defaultProps = {
  center: false,
  disabled: false,
  full: false,
  icon: '',
  size: 'medium',
  skin: 'primary',
  type: 'button',
  children: 'Catho',
  onClick: () => {},
};

Button.propTypes = {
  center: PropTypes.bool,
  disabled: PropTypes.bool,
  full: PropTypes.bool,
  /** Icon name. The full catalogue can be found
   * [here](/?selectedKind=1.%20Foundation&selectedStory=Icons) */
  icon: PropTypes.string,
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'big', 'xbig']),
  skin: PropTypes.oneOf(['primary', 'secondary', 'action']),
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export default Button;
