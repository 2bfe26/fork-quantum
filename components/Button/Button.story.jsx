import React from 'react';
import { storiesOf } from '@storybook/react';
import HowToImport from '../../.storybook/decorators/HowToImport';
import Button from './Button';

const stories = storiesOf('Button', module);
stories
  .addDecorator(HowToImport)
  .add('Button', () => (
    <Button>Click me!</Button>
  ));
