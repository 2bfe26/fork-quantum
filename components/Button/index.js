import { Button, ButtonSimple } from './Button';

export { Button, ButtonSimple };
