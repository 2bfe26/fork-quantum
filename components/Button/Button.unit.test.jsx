import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button component', () => {
  it('Should match the snapshot', () => {
    expect(renderer.create(<Button>Text</Button>).toJSON()).toMatchSnapshot();
  });

  it('when full prop is set', () => {
    expect(
      renderer.create(<Button full>Full</Button>).toJSON(),
    ).toMatchSnapshot();
  });

  it('when center prop is set', () => {
    expect(
      renderer.create(<Button center>Center</Button>).toJSON(),
    ).toMatchSnapshot();
  });

  describe('when there is a skin set', () => {
    it('should match secondary snapshot', () => {
      const primary = <Button skin="primary">primary</Button>;
      const secondary = <Button skin="secondary">secondary</Button>;
      const action = <Button skin="action">action</Button>;
      const cancel = <Button skin="cancel">cancel</Button>;
      const modal = <Button skin="modal">modal</Button>;

      expect(renderer.create(primary).toJSON()).toMatchSnapshot();
      expect(renderer.create(secondary).toJSON()).toMatchSnapshot();
      expect(renderer.create(action).toJSON()).toMatchSnapshot();
      expect(renderer.create(cancel).toJSON()).toMatchSnapshot();
      expect(renderer.create(modal).toJSON()).toMatchSnapshot();
    });
  });

  describe('when there is a type set', () => {
    it('should match secondary snapshot', () => {
      const button = <Button type="button">button</Button>;
      const reset = <Button type="reset">reset</Button>;
      const submit = <Button type="submit">submit</Button>;

      expect(renderer.create(button).toJSON()).toMatchSnapshot();
      expect(renderer.create(reset).toJSON()).toMatchSnapshot();
      expect(renderer.create(submit).toJSON()).toMatchSnapshot();
    });
  });

  describe('with an icon', () => {
    it('should match secondary snapshot', () => {
      const search = <Button icon="search">Search</Button>;
      const sync = <Button icon="sync">Sync</Button>;
      const sms = <Button icon="sms">Message</Button>;
      const camera = <Button icon="camera">Screenshot</Button>;

      expect(renderer.create(search).toJSON()).toMatchSnapshot();
      expect(renderer.create(sync).toJSON()).toMatchSnapshot();
      expect(renderer.create(sms).toJSON()).toMatchSnapshot();
      expect(renderer.create(camera).toJSON()).toMatchSnapshot();
    });
  });
});
