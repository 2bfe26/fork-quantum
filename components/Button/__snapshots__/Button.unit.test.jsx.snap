// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  Text
</button>
`;

exports[`Button component when center prop is set 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  margin-left: auto;
  margin-right: auto;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  Center
</button>
`;

exports[`Button component when full prop is set 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  width: 100%;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  Full
</button>
`;

exports[`Button component when there is a skin set should match secondary snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  primary
</button>
`;

exports[`Button component when there is a skin set should match secondary snapshot 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #FFFFFF;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #1355d0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #e5edfc;
  border-color: #1355d0;
  color: #1355d0;
}

.c0:focus {
  box-shadow: 0 2px 4px 0 rgba(19,85,208,0.5);
  background-color: #FFFFFF;
  border-color: #1355d0;
  color: #1355d0;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #e5edfc;
  border-color: #1355d0;
  color: #1355d0;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  secondary
</button>
`;

exports[`Button component when there is a skin set should match secondary snapshot 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #de0059;
  border: 1.5px solid #de0059;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #ab0045;
  border-color: #ab0045;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 #de0059;
  background-color: #de0059;
  border-color: #de0059;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #ab0045;
  border-color: #ab0045;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  action
</button>
`;

exports[`Button component when there is a skin set should match secondary snapshot 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: transparent;
  border: 1.5px solid transparent;
  box-shadow: none;
  color: #1355d0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border-radius: unset;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: none;
  background-color: undefined;
  border-color: undefined;
  color: #1355d0;
}

.c0:focus {
  box-shadow: none;
  background-color: transparent;
  border-color: transparent;
  color: #1355d0;
}

.c0:active {
  box-shadow: none;
  background-color: undefined;
  border-color: undefined;
  color: #1355d0;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  action
</button>
`;

exports[`Button component when there is a type set should match secondary snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  
  button
</button>
`;

exports[`Button component when there is a type set should match secondary snapshot 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="reset"
>
  
  reset
</button>
`;

exports[`Button component when there is a type set should match secondary snapshot 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="submit"
>
  
  submit
</button>
`;

exports[`Button component with an icon should match secondary snapshot 1`] = `
.c2 {
  margin-right: 5px;
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c3 .c1 {
  font-size: 24px;
}

.c4 .c1 {
  font-size: 24px;
}

.c5 .c1 {
  font-size: 24px;
}

.c6 .c1 {
  font-size: 24px;
}

.c7 .c1 {
  font-size: 24px;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  <span
    aria-hidden="true"
    className="material-icons MuiIcon-root-1 c1 c2"
    size="medium"
    style={
      Object {
        "color": "",
      }
    }
  >
    search
  </span>
  Search
</button>
`;

exports[`Button component with an icon should match secondary snapshot 2`] = `
.c2 {
  margin-right: 5px;
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c3 .c1 {
  font-size: 24px;
}

.c4 .c1 {
  font-size: 24px;
}

.c5 .c1 {
  font-size: 24px;
}

.c6 .c1 {
  font-size: 24px;
}

.c7 .c1 {
  font-size: 24px;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  <span
    aria-hidden="true"
    className="material-icons MuiIcon-root-1 c1 c2"
    size="medium"
    style={
      Object {
        "color": "",
      }
    }
  >
    sync
  </span>
  Sync
</button>
`;

exports[`Button component with an icon should match secondary snapshot 3`] = `
.c2 {
  margin-right: 5px;
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c3 .c1 {
  font-size: 24px;
}

.c4 .c1 {
  font-size: 24px;
}

.c5 .c1 {
  font-size: 24px;
}

.c6 .c1 {
  font-size: 24px;
}

.c7 .c1 {
  font-size: 24px;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  <span
    aria-hidden="true"
    className="material-icons MuiIcon-root-1 c1 c2"
    size="medium"
    style={
      Object {
        "color": "",
      }
    }
  >
    sms
  </span>
  Message
</button>
`;

exports[`Button component with an icon should match secondary snapshot 4`] = `
.c2 {
  margin-right: 5px;
  pointer-events: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  font-size: 16px;
  padding: 0 16px;
  height: 40px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  cursor: pointer;
  background-color: #1355d0;
  border: 1.5px solid #1355d0;
  box-shadow: 0 2px 4px 0 #cccccc;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
}

.c0 .c1 {
  font-size: 24px;
}

.c0:hover {
  box-shadow: 0 2px 4px 0 #cccccc;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c0:focus {
  box-shadow: 0 2px 6px 0 rgba(19,85,208,0.5);
  background-color: #1355d0;
  border-color: #1355d0;
  color: #FFFFFF;
}

.c0:active {
  box-shadow: 0 2px 4px 0 #4c4c4c;
  background-color: #002f7b;
  border-color: #002f7b;
  color: #FFFFFF;
}

.c3 .c1 {
  font-size: 24px;
}

.c4 .c1 {
  font-size: 24px;
}

.c5 .c1 {
  font-size: 24px;
}

.c6 .c1 {
  font-size: 24px;
}

.c7 .c1 {
  font-size: 24px;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
  size="medium"
  type="button"
>
  <span
    aria-hidden="true"
    className="material-icons MuiIcon-root-1 c1 c2"
    size="medium"
    style={
      Object {
        "color": "",
      }
    }
  >
    camera
  </span>
  Screenshot
</button>
`;
